code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -61,76 +68,109 @@
     AccountDao accountDao;
 
     @Inject
+    InstanceDao instanceDao;
+
+    @Inject
     LockManager lockManager;
 
     @SuppressWarnings("unchecked")
     @Override
     public HandlerResult handle(ProcessState state, ProcessInstance process) {
-        if ( !MANAGE_NONRANCHER_CONTAINERS.get() ) {
+        if (!MANAGE_NONRANCHER_CONTAINERS.get()) {
             return null;
         }
 
-        ContainerEvent event = (ContainerEvent)state.getResource();
-        Map<String, Object> inspect = (Map<String, Object>)DataUtils.getFields(event).get(FIELD_DOCKER_INSPECT);
+        final ContainerEvent event = (ContainerEvent)state.getResource();
 
-        String rancherUuid = getRancherUuidLabel(inspect);
-        if ( rancherUuid != null ) {
-            return null;
-        }
+        HandlerResult result = lockManager.lock(new ContainerEventInstanceLock(event.getAccountId(), event.getExternalId()), new LockCallback<HandlerResult>() {
+            @Override
+            public HandlerResult doWithLock() {
+                Map<String, Object> inspect = (Map<String, Object>)DataUtils.getFields(event).get(FIELD_DOCKER_INSPECT);
```

Comment:
Looked into making this change, but CollectionUtils.toMap(Object) returns an empty map when the input is null. The input in this case the docker inspect and while a a null value is valid (no inspect is available for destroy events), an empty non-null map is not and it causes NPEs further down. And I'd rather just check whether or not inspect is null than its children as well.